server/bin/pbench-unpack-tarballs.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+from pathlib import Path
+import sys
+
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.server import PbenchServerConfig
+from pbench.server.database import init_db
+from pbench.server.unpack_tarballs import UnpackTarballs
+
+_NAME_ = "pbench-unpack-tarballs"
+
+# The link source for the operation of this script.
+_linksrc = "TO-UNPACK"
+
+
+def main(cfg_name):
+    if not cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable.",
+            file=sys.stderr,
+        )
+        return 2
+
+    try:
+        config = PbenchServerConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    # We're going to need the Postgres DB to track dataset state, so setup
+    # DB access.
+    init_db(config, logger)
+
+    try:
+        BUCKET = str(sys.argv[1])
+    except Exception as exc:
+        logger.error("Bucket not Defined: ", exc)
+        return 1
+
+    prog = Path(sys.argv[0]).name
+
+    lowerbound, upperbound = 0, float("inf")
```

Comment:
I keep coming back to this because (even though Python has no problem with it) I really dislike implicitly mixing int and float operations. Unfortunately Python doesn't have an ingrained concept of "maximum integer" or "maximum file size", and `float("inf")` (or the arguably more efficient `math.inf`) can't be converted to integer.

One "clean" alternative would be to deal with this entirely in float with `lowerbound, upperbound = 0.0, float("inf")` and `float(Path(tarball).stat().st_size)`

Another would be to use `sys.maxsize` (which is technically the maximum number of elements a list can contain, 9223372036854775807) as a proxy for "a lot of [mega]bytes" here... but (despite the risk of "imprecision" at large sizes), going to consistent floating point would probably be "neater".

And if nobody else is bothered by this, I'll hold my nose and resolve the conversation. 😆

Reply:
I think I would be bothered more by converting the interface to use `float`'s _instead_ of `int`'s than I am by us passing in `float("inf")` where we are otherwise expecting an `int`.  Using `sys.maxsize` instead of `float("inf")` is an elegant hack which I would be willing to accept, but I think that the fact that Python is willing to accept the semantic equivalent of "number" as a type is something that we should just go with.  That is, Python doesn't care whether we pass in `int`'s or `float`'s -- they are both "numbers" and Python is willing to deal with them for us -- so, I'm inclined to accept that approach...although, if we agree to that, then it would be appropriate to modify the type hint for the function to `Union[int, float]`.

Reply:
Currently the min is `int` and the `max` is float, which is rather bizarre anyway. Although I'm not crazy about mixing them, if we're going to then we should change 55-57 below to result in a `float`...

Reply:
I don't like `Union[int, float]` particularly ... but, on the other hand, this is essentially like a Java generic since Python is perfectly happy interchanging the two types. (Though there are still some implicit casts where you need to be careful, and I'd much prefer being explicit...)

Reply:
This should be `lowerbound, upperbound = 0.0, float("inf")` so that both defaults are floating point, which matches your new type hints.

I'm still also inclined to think that we should be using `float()` explicitly on the `config.get` values if they're truthy: e.g., 

```python
        if lowerbound:
            lowerbound = float(lowerbound) * 1024.0 * 1024.0
```

"Just to be safe", because if `lowerbound` is `"a"`, then `lowerbound *= 1024 * 1024` is going to be a string of a million "a"s ... whereas with the `float(lowerbound)` it'll raise a `ValueError`.